src/dist/replication/lib/replica_http_service.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include <nlohmann/json.hpp>
+#include <fmt/format.h>
+#include "replica_http_service.h"
+#include "duplication/duplication_sync_timer.h"
+
+namespace dsn {
+namespace replication {
+
+void replica_http_service::query_duplication_handler(const http_request &req, http_response &resp)
+{
+    if (!_stub->_duplication_sync_timer) {
```

Comment:
In the future refactor, we should add a pure virtual validate function in base class `http_service`, and it will be called before this 'handler' been called.